### DISCUSSION TITLE
Error

### DISCUSSION THREAD

**Discussion by edythneska** [SERVIR-WA/GALUP#407] [General]
Good day,
I'm got an error when doing the rasterization under the "Measure Distance to each cluster". The rasterize option is 3 there .

Attached is an image of the error. Thank you
![Screenshot (22)](https://user-images.githubusercontent.com/86968514/163675300-57d63b20-84d6-4dd3-9eeb-fcc2f3760666.png)
u

**Comment by chjch** [comment]
Hi @edythneska can you please show the parameters tab and let us know what was the parameters you were using. This error message is too generic.

**Reply by edythneska** [reply]
Hello @chjch. Okay
And which of the rasterize options should be used please because there are 3 there
![Screenshot (23)](https://user-images.githubusercontent.com/86968514/163723383-b8d6d8ed-1dcb-4db4-a565-70cb763ac034.png)
e

**Reply by chjch** [reply]
The one that you are choosing, i.e., the GDAL function, is indeed the one to use.

**Comment by chjch** [comment]
For the "Output raster size units", instead of "Pixels", you should choose the option of "Georeferenced units."

**Reply by chjch** [reply]
the horizontal and vertical resolution are 100 meter, which is the same size as the land cover and drainage cell size.

**Reply by chjch** [reply]
@eidealex is there anything else needs to be set, besides what I mentioned above?

**Reply by eidealex** [reply]
That should do it @chjch. If you are trying to rasterize the urban clusters, then you can also set the burn-in value to be the CLUSTER_ID. Did that work for you @edythneska?

**Reply by edythneska** [reply]
It worked now after setting out the parameters. But my output is also empt
![Screenshot (24)](https://user-images.githubusercontent.com/86968514/163735486-d6ad79fb-91a7-4327-af71-eeb5e4917d2f.png)
y

**Comment by Achidago** [comment]
![Screenshot (106)](https://user-images.githubusercontent.com/85199074/163734660-842306b4-7dd8-4474-8d8d-edb1bd562427.png)
![Screenshot (107)](https://user-images.githubusercontent.com/85199074/163734662-8a8189e2-9236-4a82-ab10-d75b334d3a94.png)
![Screenshot (108)](https://user-images.githubusercontent.com/85199074/163734671-7af244e1-2186-44de-b3d0-59a7b181d5fb.png)
I am having similar challenges. The output is empty

**Reply by eidealex** [reply]
Hi @Achidago, have you used the DBSCAN Clustering Tool on your extracted urban points yet? Once you have your urban point clusters, you can use the rasterize tool to convert them to raster.

**Reply by edythneska** [reply]
My output is also empty

**Reply by edythneska** [reply]
I tried the Proximity tool and I got thi
![Screenshot (26)](https://user-images.githubusercontent.com/86968514/163736010-1b06799f-6b68-42bd-920f-86ae9e65aece.png)
s

**Reply by Achidago** [reply]
> Hi @Achidago, have you used the DBSCAN Clustering Tool on your extracted urban points yet? Once you have your urban point clusters, you can use the rasterize tool to convert them to raster.

I will do this. Thanks

**Comment by chjch** [comment]
@eidealex  what was the parameters we used for DBSCAN? I think the error is due to incorrect result of DBSCAN?

**Reply by eidealex** [reply]
![DBSCANparameters](https://user-images.githubusercontent.com/97194321/163736567-c35d9359-f557-4665-96c4-ded953618045.jpg)
Here are the parameters that we used for determining the urban clusters.

**Comment by chjch** [comment]
@Achidago @edythneska Please make sure urban clusters are properly created in the DBSCAN. To verify that, you can symbolize the result based on "CLUSTER_ID". Then, use **extract by attribute** to filter the first cluster, i.e., `CLUSTER_ID = 1`, then Rasterize the first cluster, then perform a _Proximity_. Do this for all urban clusters.

**Reply by edythneska** [reply]
@chjch I have tried the whole process again from the beginning but its still the same result.
But I noticed that with the extract by attribute parameters a number has to be put under the VALUE because each point has a value when the attribute table is opened. 
When a number is typed, the points with that value are seen but when no number is typed, no result is seen.

Could that also be a reason?
![Screenshot (27)](https://user-images.githubusercontent.com/86968514/163808171-00c30dbb-29ab-4633-b06f-b1218b3c9fcf.png)

This 2nd image is as a result of when I used the values 1 and 2 respectively when doing the extract by attribute
VALUE = 1, brown points
VALUE = 2, purple points
![Screenshot (36)](https://user-images.githubusercontent.com/86968514/163808646-b5dfe5a5-2da7-4859-a554-42ad1b633290.png)

**Reply by chjch** [reply]
@edythneska yes. That is correct. Extract by attribute only works when you specify a valid condition. In this case, you should try CLUSTER_ID = 1, instead of using the "VALUE" field. So, the idea is you isolate the first cluster by using this function.

**Reply by edythneska** [reply]
Okay, but I am not getting the CLUSTER_ID field. The only field I'm seeing is the VALUE field. This is the 1st extract by attribute after the Raster pixels to points

**Reply by chjch** [reply]
The CLUSTER_ID is created by the DBSCAN clustering tool. As @eidealex showed in the [image](https://github.com/SERVIR-WA/GALUP/discussions/407#discussioncomment-2583346), CLUSTER_ID is the default "Cluster field name."

**Reply by edythneska** [reply]
Okay. I'll try the process again and get back to you

Thank you

**Comment by edythneska** [comment]
@chjch This is my rasterization output I got and the parameters I used
![Screenshot (40)](https://user-images.githubusercontent.com/86968514/163817915-62a35c6f-291c-4f05-8f8a-7b81635a4430.png)

**Reply by chjch** [reply]
@edythneska  Can you please change "A fiexed value to burn [optional]" to 0? 0 is the default option and you shouldn't change this, since changing it to 1 will not use the CLUSTER_ID field anymore. Yours looks like it is rasterized using everything rather than points in cluster 1. Your output should only be a few pixels that are the points in cluster 1.

**Reply by edythneska** [reply]
Ohk

**Comment by EmmanuelElormDelanyo** [comment]
This is my rasterized image
![rasterized](https://user-images.githubusercontent.com/93537360/164211970-306a7bcf-128d-4e7f-9704-c63982c2636d.jpg)

This are the parameters
![parameters for rasterized](https://user-images.githubusercontent.com/93537360/164212195-15c692fc-d08e-4360-8ca5-8d4c9250699c.jpg)

**Reply by EmmanuelElormDelanyo** [reply]
kindly let me know if am on the right channel.

**Reply by EmmanuelElormDelanyo** [reply]
A. i rasterized but for cluster 2, but it rasterized for almost all the points; same for cluster 1 and cluster 2.
![rasterized2](https://user-images.githubusercontent.com/93537360/164262160-d2baa3c7-f8ca-476c-a13e-abc9d156bc33.jpg)

parameters
![parameters_rasterized2](https://user-images.githubusercontent.com/93537360/164262255-90c64af0-0aeb-40f5-982a-b05d2b8f8896.jpg)

B.my attempt to rasterize cluster 10 failed
parameters
![failed_rasterize10(parameters)](https://user-images.githubusercontent.com/93537360/164264395-7d074ab3-e6c8-431d-baa9-b9b77f169c60.jpg)
log
![failed_rasterize10(log)](https://user-images.githubusercontent.com/93537360/164264459-80467422-e920-455d-9af7-e795111a8f1c.jpg)

**Reply by chjch** [reply]
Hi @EmmanuelElormDelanyo, it's good to see that you are moving on to the AAC district for exercise. The cluster should be based on urban/built-up pixels, which corresponds to **value 6** in the raster. After you extract these pixels as vector points, then you can run DBSCAN clustering. I also want to point out that the video demos are available now, please follow them and see if they can clarify some of your confusion.

**Reply by chjch** [reply]
> This is my rasterized image ![rasterized](https://user-images.githubusercontent.com/93537360/164211970-306a7bcf-128d-4e7f-9704-c63982c2636d.jpg)
> 
> This are the parameters ![parameters for rasterized](https://user-images.githubusercontent.com/93537360/164212195-15c692fc-d08e-4360-8ca5-8d4c9250699c.jpg)

This looks correct.

**Comment by DanielBoateng** [comment]
This what i had for the proximity.
![Screenshot (51)](https://user-images.githubusercontent.com/93519966/164212856-14f21448-92b5-4fe7-ab35-408bee08d908.png)

This are the parameters

![Screenshot (52)](https://user-images.githubusercontent.com/93519966/164213074-c9a94386-7e42-4c42-b774-a7a01f2981b7.png)

**Reply by chjch** [reply]
@DanielBoateng It seems like you forgot to set the **Output extent**. Please use the base ISDA_MGRS raster for the extent.

**Comment by DanielBoateng** [comment]
When i opened the raster calculator the clusters are  not showing.
![Screenshot (54)](https://user-images.githubusercontent.com/93519966/164278879-ac2dbdbe-f1a1-4d92-89bd-d69f46b0fa86.png)

**Reply by chjch** [reply]
@DanielBoateng It should be under the Layers. Also, you need to perform proximity of individual clusters before using the raster calculator tool. You can follow the video demos that are just posted in the training. Thanks!

**Comment by Achidago** [comment]
![Screenshot (110)](https://user-images.githubusercontent.com/85199074/164574927-4aad87bd-e08b-490c-ab4d-177b4e9dd18e.png)
![Screenshot (111)](https://user-images.githubusercontent.com/85199074/164574694-8a3f66c5-a76b-4249-be96-03ca91e52783.png)
![Screenshot (112)](https://user-images.githubusercontent.com/85199074/164574730-de6f730f-66b7-4e5d-9636-54a7c5fe7ec6.png)
![Screenshot (113)](https://user-images.githubusercontent.com/85199074/164574760-d32e61ef-dd9f-4567-966a-245fdeac3517.png)
Any reason why i'm having this output? Kindly assist
